File: TarjetaEstilosSpring/src/main/java/pe/tarjetaestilos/models/SocialNetworks.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.tarjetaestilos.models;

/**
 *
 * @author julito
 */
import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.Where;
import java.time.ZonedDateTime;
import java.util.ArrayList;
import java.util.List;
import org.hibernate.annotations.NotFound;
import org.hibernate.annotations.NotFoundAction;
import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
@Table(name = "social_networks")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Where(clause = "deleted_at IS NULL")
public class SocialNetworks {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String icono;
    private String title;

    @Column(name = "created_at")
    private ZonedDateTime createdAt;

    @Column(name = "updated_at")
    private ZonedDateTime updatedAt;

    @Column(name = "deleted_at")
    private ZonedDateTime deletedAt;

    @OneToMany(mappedBy = "socialNetworks")
    @JsonIgnore
    @ToString.Exclude
    private List<AffiliatesNetworks> affiliatesNetworks = new ArrayList<>();
    
}

File: TarjetaEstilosVueDasboard/src/models/SocialNetworks.ts
class SocialNetworks{
    title:string|null;
    icono:string|null;
    created_at:Date|null;
    updated_at:Date|null;
    deleted_at:Date|null;
    id:number|null;

    constructor(
        title:string|null,
        icono:string|null,
        created_at:Date|null, updated_at:Date|null, deleted_at:Date|null, id:number|null=null
    ){
        this.title = title;
        this.icono = icono;
        this.created_at = created_at;
        this.updated_at = updated_at;
        this.deleted_at = deleted_at;
        this.id = id;
    }
};
export default SocialNetworks;